File: modules/algorithms-and-ds/modules/data-structures/linear/src/singly_linked_list.py
class Node:

	# Function (constructor) to initialise the node object.
	def __init__(self, data):
		self.data = data # Assign data.
		self.next = None # Initialize next as NULL (None in Python).


# Linked List class contains a Node object.
class LinkedList:

	# Function (constructor) to initialize the head.
	def __init__(self):
		self.head = None

	# This function prints contents of linked list
	# starting from head
	def printList(self):
		temp = self.head
		while (temp):
			print(temp.data)
			temp = temp.next


# Driver's code.
if __name__ == '__main__':

	# Start with the empty list
	llist = LinkedList()

	llist.head = Node(1) # assign data to first node.
	second = Node(2)     # assign data to second node.
	third = Node(3)      # assign data to third node.

	llist.head.next = second # Link first node with second.
	second.next = third      # Link second node with the third node.

	llist.printList() # Print Node's values.

File: modules/algorithms-and-ds/modules/data-structures/linear/src/singly_linked_list.cpp
#include <iostream>
using namespace std;

// Node class representation.
class Node {
public:
    int data;
    Node *next;
};

void printList(Node* n); // Function prototype.

// Driver's code.
int main()
{
    Node *head = NULL;
    Node *second = NULL;
    Node *third = NULL;

    // allocate 3 nodes in the heap.
    head = new Node();
    second = new Node();
    third = new Node();

    head->data = 1;       // assign data in first node.
    head->next = second;  // Link first node with second.

    second->data = 2;     // assign data to second node.
    second->next = third; // Link second node with third.

    third->data = 3;      // assign data to third node.
    third->next = NULL;   // Pointer next to NULL.

    // Function call
    printList(head);

    return 0;
}

// This function prints contents of linked list
void printList(Node* node_n)
{
    while (node_n != NULL)
    {
        cout << node_n->data << " ";
        node_n = node_n->next;
    }
}
